### DISCUSSION TITLE
Forking and License

### DISCUSSION THREAD

**Discussion by Veerhan-glitch** [community/community#39512] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi i recently when i thought of adding a mit license to all of my repos and saw that someone had forked my repo (made some changes to the code (did not push them)) and added their own license to it. also there is no mention to my original work (except the part of GitHub showing that it was forked from my repo).  i had also read recently that if someone forks ur repo they are not allowed change the licenses, so should i ask them to remove their license and add mine instead or its all fair as my license wasn't present when the fork was made.

**Comment by airtower-luna** [comment, marked as answer]
> Hi i recently when i thought of adding a mit license to all of my repos and saw that someone had forked my repo (made some changes to the code (did not push them)) and added their own license to it.

At a glance (not legal advice!) that smells like a copyright violation. "No license" means you don't grant others any rights, see https://choosealicense.com/no-permission/ for good summary. Note that [GitHub's ToS say that by creating a public repositories you allow others to fork it](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#5-license-grant-to-other-users), but that doesn't waive other rights (for example the one to attribution).

> should i ask them to remove their license and add mine instead

That is a completely reasonable request. Keep in mind that if they add their own code they may want to *add* (not replace) their own attribution. And depending on the license you choose they could use a different license for extensions/modifications they make.

**Reply by Veerhan-glitch** [reply]
Hmm I think it is as they have also uploaded it as their own vs theme (although is it as u don't gain anything commercial by vsc extention) 


Also
So acording to my understanding  of MIT license, If I add one people are allowed to use my code as their own unless it's commercial right?

**Reply by airtower-luna** [reply]
Read the actual license, it's pretty short: https://choosealicense.com/licenses/mit/

But as a summary I'd say that claiming the code is their own is pretty much the one thing they are *not* allowed to do. Commercial use *is* allowed by the MIT license.

**Reply by Veerhan-glitch** [reply]
Hmm got it thx for answering.
